### PR TITLE
feat: add OrganizationSwitchModal

### LINKED ISSUE #793: Use right organization id for all requests
### Describe the chore

Use right organization id for all requests, currently we are defaulting to a specific id instead of using the one of the current organization

### PR DESCRIPTION
**Pull request by MaxSchaefer**
## Which issues does this pull request close?
closes #793

### CODE REVIEW

**Comment by MaxSchaefer** [comment]
@DasProffi @00100200345

File: tasks/components/layout/PageWithHeader.tsx [hunk 1]
```diff
@@ -24,10 +28,13 @@ export const PageWithHeader = ({
   crumbs
 }: PropsWithChildren<PageWithHeaderProps>) => {
   const { user } = useAuth()
+  const [isOrganizationSwitchModalOpen, setOrganizationSwitchModalOpen] = useState(false)
+  const [organization, setOrganization] = useState<OrganizationDTO>()
 
   if (!user) return null
 
   const feedbackButton = <FeedbackButton/>
+  const organizationName = (organization?.shortName && <Span onClick={() => setOrganization(undefined)}>{organization.shortName}</Span>)
```
Comment (use-to) [inline tasks/components/layout/PageWithHeader.tsx:37]:
```suggestion
  const organizationName = (organization?.shortName && <Span onClick={() => setOrganization(undefined)} className={tw('cursor-pointer hover:cursor-pointer')}>{organization.shortName}</Span>)
```

**Review by use-to — CHANGES_REQUESTED** [review]

**Comment by use-to** [comment]
@MaxSchaefer This is still a draft. Why?

**Review by use-to — CHANGES_REQUESTED** [review]
The loaded view disappears, when clicking. This should not be the case.																																																																																																																																																																																																																																																																																																																																																																																																																																																																												
																																																																																																																																		
[background-reset.webm](https://github.com/helpwave/web/assets/13922835/33062a7d-a893-41cc-a925-45f9861d639f)

**Review by use-to — CHANGES_REQUESTED** [review]
Several issues need addressing in this code. Firstly, the complete page reload is cumbersome. Can we explore alternatives?
Secondly, when opening the modal, the current selection disappears along with the button that was just clicked.

File: tasks/hooks/useOrganization.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <OrganizationContext.Provider value={{ organization, setOrganization, setOrganizationId }}>
+      <OrganizationSwitchModal
+        id="organization-switch-modal"
+        isOpen={isOrganizationSwitchModalOpen}
+        organizations={organizations}
+        onDone={(organization) => {
+          setOrganization(organization)
+
+          // A manual switch of the organization will result in a page reload to refetch all the queries
+          window.location.reload()
```
Comment (MaxSchaefer) [inline tasks/hooks/useOrganization.tsx:83]: @JannikWibker the `window.location.reload()` is far from okay. I've tried to use `queryClient.clear()` and `queryClient.refechQueries()`, no effects. Can you take a look?

**Review by use-to — CHANGES_REQUESTED** [review]
image

**Review by use-to — CHANGES_REQUESTED** [review]
Bug still present. See image of the review before.

**Comment by MaxSchaefer** [comment]
Preview didnt catched up ...
Ready on preview.

**Review by use-to — APPROVED** [review]
Yes, my fault. Didn't recognize that.

**Pull request merged**